src/main/java/org/apache/sling/xss/impl/HtmlSanitizer.java

```diff
@@ -0,0 +1,88 @@
+ ~ under the License.
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
+package org.apache.sling.xss.impl;
+
+import java.lang.reflect.Field;
+
+import org.apache.sling.xss.impl.xml.AntiSamyPolicy;
+import org.owasp.html.DynamicAttributesSanitizerPolicy;
+import org.owasp.html.Handler;
+import org.owasp.html.HtmlStreamEventReceiver;
+import org.owasp.html.HtmlStreamRenderer;
+import org.owasp.html.PolicyFactory;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+
+public class HtmlSanitizer {
+
+    private AntiSamyPolicyAdapter customPolicy;
+    private ImmutableMap policies;
+    private ImmutableSet<String> textContainers;
+
+    public HtmlSanitizer(AntiSamyPolicy policy) {
+        this.customPolicy = new AntiSamyPolicyAdapter(policy);
+        policies = reflectionGetPolicies(customPolicy.getHtmlCleanerPolicyFactory());
+        textContainers = reflectionGetTextContainers(customPolicy.getHtmlCleanerPolicyFactory());
+    }
+
+    public SanitizedResult scan(String taintedHTML) {
+        StringBuilder sb = new StringBuilder(taintedHTML.length());
+        HtmlStreamEventReceiver out = HtmlStreamRenderer.create(sb, Handler.DO_NOTHING);
+        DynamicAttributesSanitizerPolicy dynamicPolicy = new DynamicAttributesSanitizerPolicy(out, policies,
+                textContainers, customPolicy.getDynamicAttributesPolicyMap(), customPolicy.getOnInvalidRemoveTagList());
+
+        org.owasp.html.HtmlSanitizer.sanitize(taintedHTML, dynamicPolicy,
+                customPolicy.getCssValidator().newStyleTagProcessor());
+        return new SanitizedResult(sb.toString(), dynamicPolicy.getNumberOfErrors());
+    }
+
+    private ImmutableSet<String> reflectionGetTextContainers(PolicyFactory policyFactory) {
+        Class<?> c = policyFactory.getClass();
+        try {
+            Field field = c.getDeclaredField("textContainers");
+            field.setAccessible(true);
+            return (ImmutableSet<String>) field.get(policyFactory);
+        } catch (NoSuchFieldException | SecurityException | IllegalAccessException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private ImmutableMap reflectionGetPolicies(PolicyFactory policyFactory) {
+        Class<?> c = policyFactory.getClass();
+        try {
+            Field field = c.getDeclaredField("policies");
+            field.setAccessible(true);
+            return (ImmutableMap) field.get(policyFactory);
+        } catch (NoSuchFieldException | SecurityException | IllegalAccessException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public String scan(String taintedHTML, AntiSamyPolicy policy) throws Exception {
+        if (taintedHTML == null) {
+            throw new Exception("Null html input");
+        }
+
+        if (policy == null) {
+            throw new Exception("No policy loaded");
+        }
+        return "safeHTML";
```

Comment:
What is the use case for this methods which always returns the literal?

Reply:
Resolved: b240210efc73b6a2898304dfd686a3f11da549ea